src/Module/AbstractBaseModule.php

```diff
@@ -238,9 +291,16 @@ protected function _initModuleEvents($eventManager, $eventFactory)
      */
     protected function _setupContainer($config, $services)
     {
+        $configData        = $this->_createCompositeContainer([
+            [$this->_getKey() => $this->_getConfig()],
```

Comment:
This looks like the config injected into the module will implicitly be under the module's key. Is this really a good requirement?

Reply:
It being under the module key is one possible solution to be problem of shadowing. A module's injected config will have the `key` config all of the time and `dependencies` config for about 50% of the time. Any module this is setup later will override that. Being under the module key prevents that.

Whether it should be explicit or implicit is up for debate. But my thoughts are this: the module receives config for _itself_; to know its key and deps and to be able to have custom information for itself internally, such as the paths to the services and config files. The config is attached to the setup container for convenience's sake, to allow external objects to read it. So explicitly injecting the config under a module key makes less sense, at least to me from this POV.